internal/configs/test_file.go

```diff
@@ -537,6 +662,143 @@ func decodeTestRunOptionsBlock(block *hcl.Block) (*TestRunOptions, hcl.Diagnosti
 	return &opts, diags
 }
 
+func decodeOverrideResourceBlock(block *hcl.Block, mode addrs.ResourceMode) (*OverrideResource, hcl.Diagnostics) {
+	parseTarget := func(attr *hcl.Attribute) (hcl.Traversal, *addrs.ConfigResource, hcl.Diagnostics) {
+		traversal, traversalDiags := hcl.AbsTraversalForExpr(attr.Expr)
+		diags := traversalDiags
+		if traversalDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		configRes, configResDiags := addrs.ParseConfigResource(traversal)
+		diags = append(diags, configResDiags.ToHCL()...)
+		if configResDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		return traversal, &configRes, diags
+	}
+
+	res := &OverrideResource{
+		Mode: mode,
+	}
+
+	content, diags := block.Body.Content(overrideResourceBlockSchema)
+
+	if attr, exists := content.Attributes["target"]; exists {
+		target, parsed, moreDiags := parseTarget(attr)
+		res.Target, res.TargetParsed = target, parsed
+		diags = append(diags, moreDiags...)
+	}
+
+	if attr, exists := content.Attributes["values"]; exists {
+		v, moreDiags := parseObjectAttrWithNoVariables(attr)
+		res.Values, diags = v, append(diags, moreDiags...)
+	}
+
+	return res, diags
+}
+
+func decodeOverrideModuleBlock(block *hcl.Block) (*OverrideModule, hcl.Diagnostics) {
+	parseTarget := func(attr *hcl.Attribute) (hcl.Traversal, addrs.Module, hcl.Diagnostics) {
+		traversal, traversalDiags := hcl.AbsTraversalForExpr(attr.Expr)
+		diags := traversalDiags
+		if traversalDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		target, targetDiags := addrs.ParseModule(traversal)
+		diags = append(diags, targetDiags.ToHCL()...)
+		if targetDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		return traversal, target, diags
+	}
+
+	mod := &OverrideModule{}
+
+	content, diags := block.Body.Content(overrideModuleBlockSchema)
+
+	if attr, exists := content.Attributes["target"]; exists {
```

Comment:
Same as before - Isn't the `target` attribute in the block required? Feels like we should throw an error if it's missing.

Reply:
Same - Yes, it is required and we throw an error if it's missing. The check is done before this function is called.

It is marked as required in the schema: https://github.com/opentofu/opentofu/blob/override-for-testing/internal/configs/test_file.go#L893